framework/sac_rlhf.py

```diff
@@ -253,7 +255,40 @@ def load_replay_buffer(replay_buffer: ReplayBuffer, path: str):
     replay_buffer.ground_truth_rewards = buffer_data["ground_truth_rewards"]
     replay_buffer.dones = buffer_data["dones"]
 
-    logging.debug(f"Replay buffer loaded from {path}")
+    logging.info(f"Replay buffer loaded from {path}")
+
+
+def uniform_sampling(rb: ReplayBuffer, reward_net: RewardNet):
+    traj_mb1, traj_mb2 = rb.sample_trajectories()
+    return traj_mb1[0], traj_mb2[0]
+
+
+def disagreement_sampling(rb: ReplayBuffer, reward_net: RewardNet):
+    traj_mb1, traj_mb2 = rb.sample_trajectories()
+
+    disagrees = []
+    for traj_1, traj_2 in zip(traj_mb1, traj_mb2):
+        probs = []
+        for member in range(len(reward_net.ensemble)):
+            probs.append(reward_net.preference_prob_hat_member(traj_1, traj_2, member))
+        disagrees.append(np.std(probs, axis=0))
+
+    disagrees_argmax = max(enumerate(disagrees), key=lambda x: x[1])[0]
+    return traj_mb1[disagrees_argmax], traj_mb2[disagrees_argmax]
+
+
+def entropy_sampling(rb: ReplayBuffer, reward_net: RewardNet):
+    traj_mb1, traj_mb2 = rb.sample_trajectories()
+
+    entropies = []
+    for traj_1, traj_2 in zip(traj_mb1, traj_mb2):
+        probs = []
+        for member in range(len(reward_net.ensemble)):
+            probs.append(reward_net.preference_prob_hat_member(traj_1, traj_2, member))
```

Comment:
This must be `reward_net.preference_hat_entropy_member(traj_1, traj_2, member)`,